### DISCUSSION TITLE
RTX 4090 benchmarks - FLUX model

### DISCUSSION THREAD

**Discussion by belenkoksas** [Comfy-Org/ComfyUI#4571] [General]
The problem is that everyone has different configurations, and my ComfyUI setup was a mess. The FLUX model took a long time to load, but I was able to fix it.

**My PC Specifications:**
Processor: Intel i9-12900K @ 3.20 GHz
Memory: 64.0 GB (63.7 GB usable)
GPU: NVIDIA RTX 4090

**Comfy log:**
![image](https://github.com/user-attachments/assets/387dac5d-c3c4-46ef-9a27-04c551308e02)

**Goal:**
I want to see if my setup is one of the fastest in the community, using the same workflows and models.

**Workflow:** 
![ComfyUI_14258_](https://github.com/user-attachments/assets/06d8f6b7-b06d-4f7b-9525-ee838be3decd)

**Workflow Screenshot:**
![image](https://github.com/user-attachments/assets/874beb1b-f92c-41a9-bcce-41b6a3d9dadd)

**My RTX 4090 Results:**
I ran the process twice each time. The first run took longer due to loading models and other initializations:

1 run: Prompt executed in 41.14 seconds
2 etc... runs: Prompt executed in 18.42 seconds

![image](https://github.com/user-attachments/assets/5970efa6-e6f7-489f-91e5-302608892102)

**Comment by ltdrdata** [comment]
RTX 4080 / torch==2.5.0.dev20240821+cu124 / python 3.12 / Ubuntu 24.04
ComfyUI (7df42b9a2364bae68)

* weight_dtype (default) (28sec)

![image](https://github.com/user-attachments/assets/1741b9e9-8563-45d3-ad5b-578007ed8d6c)

* weight_dtype (fp8_e4m3fn) without --fast (19sec)

![image](https://github.com/user-attachments/assets/8c508f19-0e73-4179-9c1d-5e875bcadfb9)

![image](https://github.com/user-attachments/assets/49b4c04c-7954-4f81-a9e9-bd078bce49a1)

* weight_dtype (fp8_e4m3fn) with --fast (13sec)

![image](https://github.com/user-attachments/assets/d3915938-890a-43fe-9b14-78c97f01f325)

**Comment by JorgeR81** [comment]
Flux in Q8_0 format looks very similar to FP16 ( better quality than FP8 ) and may be even faster.
Definitely faster, while loading, for me. 

Probably will be even faster in the future, while generating ( https://github.com/comfyanonymous/ComfyUI/issues/4538#issuecomment-2303605586 )
 
https://github.com/city96/ComfyUI-GGUF
https://huggingface.co/city96/FLUX.1-dev-gguf/tree/main

**Comment by scpedicini** [comment]
Using the Q8 GGUF I see 15 seconds at the fastest to 17 seconds at the slowest on my RTX 4090 with Euler 20 Steps for 1024x1024 images.

<img width="300" alt="CleanShot 2024-08-25 at 15 22 26@2x" src="https://github.com/user-attachments/assets/e7dc25b9-48ab-4cb5-9efb-6668cf6534d6">

It's around 17 seconds if I'm continually changing up the prompt (I think because its having to unload/reload text encoders), for continual generation its about 15 seconds. 

Previously I was using Forge with the FP8 model where I was seeing 14 seconds.

**Comment by jepjoo** [comment]
Updating from pytorch 2.4.0+cu121 to latest 2.5.0.dev+cu124 boosted generation speed by about 10%.

1024x1024 20step now:

With 2.52 GHz at 875mV, drawing ~290W:
100%|██████████████| 20/20 [00:09<00:00,  2.09it/s]
Prompt executed in 10.01 seconds

And 2.8 GHZ at 1000mV, drawing 400W:
100%|██████████████| 20/20 [00:08<00:00,  2.29it/s]
Prompt executed in 9.07 seconds

With pytorch 2.4 I was at about 1.9t/s and 260-270W with the heavily undervolted 2.52GHz clockspeed that I normally use.

All above results are with FP8 --fast, but it seems GGUF Q8 got a similar ~10% speed bump with 2.5.0.dev+cu124.

**Comment by metamountain** [comment]
3090  using 400W :(


![image](https://github.com/user-attachments/assets/afea7458-b6ea-4cec-ae95-d68a3b11d7db)

![image](https://github.com/user-attachments/assets/91571883-1259-4e09-8445-79f7f3ddc0f6)

**Reply by l33tx0** [reply]
Run GPU at 70-75% power you'll get nearly the same performance , GPU will run cooler too.

**Reply by baekdoosixt** [reply]
> Run GPU at 70-75% power you'll get nearly the same performance , GPU will run cooler too.

Can you tell me how you do that please ?

**Reply by l33tx0** [reply]
you can use **msi afterburner** , and set the power limit to 75 percent then apply 
![image](https://github.com/user-attachments/assets/a513eac2-6720-41f1-96b7-72c6f6931abc)

**Reply by baekdoosixt** [reply]
Thanks a lot !

**Comment by cybertect74-sketch** [comment]
4090 triton+ sage attention

Model Flux prepared for dynamic VRAM loading. 11350MB Staged. 0 patches attached.
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:06<00:00,  3.16it/s]
Model AutoencodingEngine prepared for dynamic VRAM loading. 159MB Staged. 0 patches attached.
Prompt executed in 7.15 seconds


Checkpoint files will always be loaded safely.
Total VRAM 24564 MB, total RAM 64630 MB
pytorch version: 2.9.1+cu130
Enabled fp16 accumulation.
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 4090 : native
Using async weight offloading with 2 streams
Enabled pinned memory 29083.0
working around nvidia conv3d memory bug.
Using sage attention
aimdo: src/control.c:51:INFO:comfy-aimdo inited for GPU: NVIDIA GeForce RTX 4090 (VRAM: 24563 MB)
DynamicVRAM support detected and enabled
Python version: 3.13.6 (tags/v3.13.6:4e66535, Aug  6 2025, 14:36:00) [MSC v.1944 64 bit (AMD64)]
ComfyUI version: 0.14.1
ComfyUI frontend version: 1.38.14

**Comment by xXMrNidaXx** [comment]
RTX 4090 FLUX benchmarks are impressive! At RevolutionAI (https://revolutionai.io) we run image gen workloads.

**Our 4090 benchmarks (FLUX.1 dev):**

| Resolution | Steps | Time | it/s |
|------------|-------|------|------|
| 512x512 | 20 | 4s | 5.0 |
| 1024x1024 | 20 | 12s | 1.7 |
| 1024x1024 | 50 | 28s | 1.8 |

**Optimizations applied:**
- fp16 inference
- Tiled VAE decode
- xformers enabled
- CUDA graphs

**Tips for speed:**
1. Use fp16/bf16 checkpoint
2. Enable attention optimizations
3. Compile with torch.compile
4. Batch when possible

**Memory usage:**
- FLUX.1 dev: ~12GB VRAM
- With optimizations: ~10GB

Great GPU for FLUX workflows!
